pandas/core/tools/datetimes.py

```diff
@@ -904,7 +905,7 @@ def to_datetime(
 
     >>> pd.to_datetime([1, 2, 3], unit="D", origin=pd.Timestamp("1960-01-01"))
     DatetimeIndex(['1960-01-02', '1960-01-03', '1960-01-04'],
-                  dtype='datetime64[ns]', freq=None)
+                  dtype='datetime64[s]', freq=None)
```

Comment:
This makes me wonder about a potential corner case: what if the original actually has values for a lower resolution, like actual microseconds or nanoseconds?